pages/implementation/reference/javascript.mdx

```diff
@@ -0,0 +1,190 @@
+>
+> Find examples of JavaScript custom destinations for each analytics destination in <Link passHref href="/implementation/destinations"><a>this section</a></Link>.
+
+##### setAvoLogger
+
+```javascript
+Avo.setAvoLogger(avoLogger)
+```
+
+Separarte method to set custom logger
+
+> **Arguments**
+>
+> `avoLogger`: custom implementation of the logger. Can be used to disable logs. <Link passHref href="/implementation/custom-loggers"><a>More about custom loggers</a></Link>  
+
+##### setSystemProperties
+
+```javascript
+Avo.setSystemProperties(systemProperties)
+```
+
+A method to update system properties. If you provide undefined values here corresponding properties won't be updated
+
+>**Arguments**
+>
+>`systemProperties`: an object where each field represents a system propety in your tracking plan. 
+>When you define system properties in your Avo workspace you set name and type - the fileds of this object are named the same as system propertyies, in lower camel case, and you should provide corresponding types.
+
+##### Event tracking functions
+
+```javascript
+Avo.yourEventName(properties: { [anonymousUserId_], [userId_], [segmentContext_], [eventProperty0], [eventProperty1], ..., [userProperty0], [userProperty1], ... })
+```
+
+Every event you define in your tracking plan in Avo gets a function named after the event in lower camel case. The arguments of the function depend on how it's defined in your tracking plan 
+
+> **Arguments**
+>
+> `anonymousUserId_`: string, Node.js only, this argument is automatically added if corresponding setting is enabled, user to track anonymous users
+>
+> `userId_`: string
+> Node.js: added to all events, you have to either provide it or the `anonymousUserId_`
+> Web and React Native: Added if the event has the Identify User action
+>
+> `eventProperty`: type defined in the Avo tracking plan, can be string, int, long, float, bool, array and any
+> This is an event property that should be tracked with given event. Pass the value of the property to track here.
+>
+> `userProperty`: type defined in the Avo tracking plan, can be string, int, long, float, bool, array and any
+> This is a user property that should be updated with given event. Pass the value of the property to update here.
+>
+> `segmentContext_`: object, Node.js only, passed down to Segment as the Segment constext, e.g. `segment.track({..., context: context})`
+
+### Destinations
+
+You can send your data from Javascript to any event based analytics destionation, including
+
+- Amplitude
+- FacebookAnalytics
+- FullStory
+- Mixpanel
+- Mixpanel
+- Permutive
+- Segment
+- Snowplow
+- ZendeskConnect
+- Adobe Analytics
+- Apptelemetry
+- Rudderstack
+- Freshpaint
+- Posthog
+- Google Analytics 4 / Firebase Analytics
+- Heap
+- Keen
+- Kissmetrics
+- LaunchDarkly Events
+- Pendo
+- Fivetran
+- AppsFlyer
+- Braze
+- Intercom
+- A home made SDK
+- Internal API
```

Comment:
Todo: Decide how we'll structure the docs for these. For Avo managed destinations we should provide details on which method each Avo action triggers + example of custom destinations. For all non-Avo managed destinations we should provide example of a custom destination.